### PR TITLE
feat(ecs): environment files for containers in EC2 task definitions #10433

### LINKED ISSUE #10433: [ecs] Add support for the environmentFiles option in ContainerDefinition
Right now `environment` is supported, but we'd like to add environment variables in bulk using the `environmentFiles` option as described here: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/taskdef-envfiles.html

### Use Case

We'd like to store our `.env` files for a node app in a single place (S3) to avoid drift, the files are quite large and maintaining each variable with `environment` isn't feasible.

### Proposed Solution

Support the [`environmentFiles` field](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions.html#cfn-ecs-taskdefinition-containerdefinition-environmentfiles)

* [x] :wave: I may be able to implement this feature request

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by rrhodes**
Add support for the environment files prop in container definitions. This prop is supported for task definitions using the EC2 launch type for objects hosted on S3. AWS recently announced support for Fargate task definitions too, but this is not yet supported in CloudFormation.

Following @MrArnoldPalmer's suggestion, a higher level solution is presented here to allow users to apply environment files both from S3 and local disk paths, similar to functionality used in the Lambda's `Code` construct.

Closes #10433

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by MrArnoldPalmer** [comment]
@rrhodes sorry for the delay, I'm catching up from some vacation time.

What do you think about something slightly higher level? I think `environmentFiles` could accept a local file asset in addition to one already uploaded to S3. We could do this similarly to the [`Code`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda.Code.html) in lambda using the `fromAsset` static method for local files and the `fromBucket` method for files already in S3 somewhere.

This also allows us to add additional methods for handling files in other places in the future without breaking the type of the property.

Adding @SoManyHs @uttarasridhar from the ECS team as well.

**Review by MrArnoldPalmer — DISMISSED** [review]
@rrhodes sorry for the delay, I'm catching up from some vacation time.

What do you think about something slightly higher level? I think environmentFiles could accept a local file asset in addition to one already uploaded to S3. We could do this similarly to the Code in lambda using the fromAsset static method for local files and the fromBucket method for files already in S3 somewhere.

This also allows us to add additional methods for handling files in other places in the future without breaking the type of the property.

Adding @SoManyHs @uttarasridhar from the ECS team as well.

**Comment by rrhodes** [comment]
Hi @MrArnoldPalmer, that sounds like a good suggestion to me. I'm happy to explore that idea later this week and report back unless anyone in the team is eager to pick this up?

**Comment by rrhodes** [comment]
Hi @MrArnoldPalmer, this PR is ready for review again with your initial feedback addressed. Let me know your thoughts - I'll be happy to make any amendments to finish this new feature.

**Comment by rrhodes** [comment]
Hi @MrArnoldPalmer, any further feedback regarding this PR?

**Comment by rrhodes** [comment]
Hi @MrArnoldPalmer, any additional feedback on this PR?

**Comment by rrhodes** [comment]
Hi @SoManyHs, @uttarasridhar, any feedback regarding this PR? Should be a relatively small change to support Fargate too when CloudFormation supports this.

**Comment by SoManyHs** [comment]
Hi @rrhodes, sorry for the delay! I'll take a look today.

File: packages/@aws-cdk/aws-ecs/lib/environment-file.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import { IBucket, Location } from '@aws-cdk/aws-s3';
+import { Asset, AssetOptions } from '@aws-cdk/aws-s3-assets';
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * Constructs for types of environment filess
```
Comment (SoManyHs) [inline packages/@aws-cdk/aws-ecs/lib/environment-file.ts:6]:
```suggestion
 * Constructs for types of environment files
```

**Review by SoManyHs — COMMENTED** [review]
Looks great! Just one small nit but otherwise 🎉

**Review by SoManyHs — APPROVED** [review]

**Comment by rrhodes** [comment]
Hi @SoManyHs,

Thanks for the feedback! Good spot on the double `s` there. Do you have permission to approve and merge this PR when a successful build is complete?

**Comment by SoManyHs** [comment]
@mergifyio update

**Review by SoManyHs — APPROVED** [review]

**Pull request merged**